src/mpriscontroller.cpp

```diff
@@ -19,6 +17,303 @@
+
+    When set to true, the manager will only control a single player
+    service. Otherwise the manager will try to guess which player
+    to control.
+*/
+
+/*!
+    \qmlproperty string MprisController::currentService
+    \brief The service name of the currently controller player
+
+    Indicates the service name of the player currently being controlled.
+    When set, the control is moved to the indicated service, but unless
+    singleService is set, control may automatically move.
+
+    Defaults to false.
+
+    \sa MprisController::singleService
+*/
+
+/*!
+    \qmlproperty list MprisController::availableServices
+    \brief List of service names of current players
+
+    The list contains service names of all players currently registered.
+*/
+
+/*!
+    \qmlproperty list MprisController::availableClients
+    \brief List of MprisClients for current players
+    \internal
+
+    Provides a list of objects that can be used to control specific player.
+    Can be used as a model for ListView or Repeater.
+
+    Note, the life time of the returned objects is only as long as they are
+    advertised by the manager, using them after the property has changed
+    will cause a crash.
+*/
+
+/*!
+    \qmlproperty bool MprisController::canQuit
+    \qmlproperty bool MprisClient::canQuit
+    \brief Indicates whether the controlled player can be closed
+
+    When true, the quit method can be called to request closure of
+    the player.
+*/
+
+/*!
+    \qmlproperty bool MprisController::canRaise
+    \qmlproperty bool MprisClient::canRaise
+    \brief Indicates whether the controlled player can present itself
+
+    When true, the raise method can be called to request the controlled
+    player to show its interface to the user.
+*/
+
+/*!
+    \qmlproperty bool MprisController::canSetFullscreen
+    \qmlproperty bool MprisClient::canSetFullscreen
+    \brief Indicates whether the controlled player can go full screen
+
+    When true, the fullscreen property can be set to true to instruct
+    the controlled player to show its media on whole screen.
+*/
+
+/*!
+    \qmlproperty string MprisController::desktopEntry
+    \qmlproperty string MprisClient::desktopEntry
+    \brief Name of a desktop entry for the controller player
+
+    Provides a name for a desktop file that can be used to gain
+    additional information about the player.
+*/
+
+/*!
+    \qmlproperty bool MprisController::fullscreen
+    \qmlproperty bool MprisClient::fullscreen
+    \brief Indicates whether the controlled player is currently full screen
+
+    When true, the controlled player is showing its media using the whole
+    screen.
+*/
+
+/*!
+    \qmlproperty bool MprisController::hasTrackList
+    \qmlproperty bool MprisClient::hasTrackList
+    \brief Indicates whether the controlled player implements tracklist
+
+    When set to true, the controlled player implements tracklist feature.
+    This feature is currently not implemented by the library.
```

Comment:
Theoretically there could be clients that don't use the library, though.

Reply:
Yes, but neither side of this library supports it; could make it more unambiguous.